crates/router/src/core/errors/utils.rs

```diff
@@ -173,6 +173,7 @@ impl<T> ConnectorErrorExt<T> for error_stack::Result<T, errors::ConnectorError>
                 errors::ConnectorError::InvalidDataFormat { field_name } => {
                     errors::ApiErrorResponse::InvalidDataValue { field_name }
                 },
+                errors::ConnectorError::CurrencyNotSupported => errors::ApiErrorResponse::NotSupported {message: "Payment method for given currency is not configured with the given connector".into()},
```

Comment:
Let's add message and connector field for `CurrencyNotSupported` error

Reply:
Connector is sending string based errors in production.